app/init.go

```diff
@@ -0,0 +1,128 @@
+
+package app
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/app"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/commands/server"
+	"github.com/iov-one/weave/crypto"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/migration"
+	"github.com/iov-one/weave/x/cash"
+	"github.com/iov-one/weave/x/validators"
+	"github.com/iov-one/weave/x/multisig"
+	"github.com/iov-one/weave/x/currency"
+	"github.com/iov-one/weave/x/msgfee"
+	abci "github.com/tendermint/tendermint/abci/types"	
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+// GenInitOptions will produce some basic options for one rich
+// account, to use for dev mode
+//
+// You can set
+func GenInitOptions(args []string) (json.RawMessage, error) {
+	code := "DEX"
+	if len(args) > 0 {
+		code = args[0]
+	}
+
+	var addr string
+	if len(args) > 1 {
+		addr = args[1]
+	} else {
+		// if no address provided, auto-generate one
+		// and print out a recovery phrase
+		bz, phrase, err := GenerateCoinKey()
+		if err != nil {
+			return nil, err
+		}
+		addr = bz.String()
+		fmt.Println(phrase)
+	}
+
+	type (
+		dict  map[string]interface{}
+		array []interface{}
+	)
+	collectorAddr, err := hex.DecodeString("3b11c732b8fc1f09beb34031302fe2ab347c5c14")
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot hex decode collector address")
+	}
+	return json.Marshal(dict{
+		"cash": array{
+			dict{
+				"address": addr,
+				"coins": array{
+					dict{
+						"whole":  123456789,
+						"ticker": code,
+					},
+				},
+			},
+		},
+		"conf": dict{
+			"cash": cash.Configuration{
+				CollectorAddress: collectorAddr,
+				MinimalFee:       coin.Coin{Whole: 0}, // no fee
+			},
+			"migration": dict{
+				"admin": addr,
+			},
+		},
+		"initialize_schema": []dict{
+			{"pkg": "cash", "ver": 1},
+			{"pkg": "sigs", "ver": 1},
+			{"pkg": "validators", "ver": 1},
+			{"pkg": "utils", "ver": 1},
+		},
+	})
+}
+
+// GenerateApp is used to create a stub for server/start.go command
+func GenerateApp(options *server.Options) (abci.Application, error) {
+	// db goes in a subdir, but "" -> "" for memdb
+	var dbPath string
+	if options.Home != "" {
+		dbPath = filepath.Join(options.Home, "abci.db")
+	}
+
+	stack := Stack(options.MinFee)
+	application, err := Application("mycoin", stack, TxDecoder, dbPath, options.Debug)
+	if err != nil {
+		return nil, err
+	}
+
+	return DecorateApp(application, options.Logger), nil
+}
+
+// DecorateApp adds initializers and Logger to an Application
+func DecorateApp(application app.BaseApp, logger log.Logger) app.BaseApp {
+	application.WithInit(app.ChainInitializers(
```

Comment:
I see some packages do not have an initializer such as [aswap](https://github.com/iov-one/weave/tree/master/x/aswap/).  The reasons it packages like aswap does not require initialization on genesis?

Reply:
That is mostly the case. 
Sometimes we do not implement initialization because it is not yet needed. In this case, initialization from the genesis does not make much sense.